lib/code-manager.js

```diff
@@ -50,6 +50,42 @@ export function getRows(
   return normalizeString(code);
 }
 
+export function getMetadataForRow(
+  editor: atom$TextEditor,
+  start: atom$Point
+): string {
+  // `start` is a Point on the first line of the cell. The cell marker is on the
+  // previous line
+  if (start.row === 0) {
+    return "code";
+  }
+  var rowText = getRow(editor, start.row - 1);
+  if (_.includes(rowText, "md") || _.includes(rowText, "markdown")) {
+    return "markdown";
+  } else {
+    return "code";
+  }
+}
+
+export function removeCommentsMarkdownCell(
+  editor: atom$TextEditor,
+  text: string
+): string {
+  const commentStartString = getCommentStartString(editor).trimRight();
+
+  const lines = text.split("\n");
+  const editedLines = [];
+  _.forEach(lines, line => {
+    if (line.startsWith(commentStartString)) {
+      // Remove comment from start of line
+      editedLines.push(line.slice(commentStartString.length).trimLeft());
```

Comment:
Does this still allow for indentation in the markdown cell?

Reply:
You're right, I don't think it does.

Reply:
Just removing `.trimLeft()` isn't my favorite solution because then if a markdown cell looks like
```py
# %% markdown
# # Header
# **bold**
# - list
#     - indented list
```
then the text in the markdown cell in the Jupyter notebook would still have a leading space in each row, i.e.:
```
 # Header
 **bold**
 - list
     - indented list
```
I think writing with a space after the `#` is clearer in the `.py` file, while having a leading space on every line in the exported markdown cell is annoying.

I added code to dedent common leading whitespace for Markdown blocks, so that the above cell would be converted to
```
# Header
**bold**
- list
    - indented list
```